src/util.c

```diff
@@ -967,3 +968,46 @@ int utilTest(int argc, char **argv, int accurate) {
     return 0;
 }
 #endif
+
+int createPipe(int fds[2], int read_flags, int write_flags) {
+#if defined(__linux__) || defined(__FreeBSD__) 
+    int flags = O_CLOEXEC;
+
+    if ((read_flags & O_NONBLOCK) && (write_flags & O_NONBLOCK))
+        flags |= O_NONBLOCK;
+
+    int ret = pipe2(fds, flags);
+    if (ret && errno != ENOSYS) {
+        return -1;
+    } else if (ret) {
+        if (pipe(fds))
+            return -1;
+        anetCloexec(fds[0]);
+        anetCloexec(fds[1]);
+    }
+
+    if (!ret && (flags & O_NONBLOCK)) {
+        return 0;
+    }
+#else
+    if (pipe(fds))
+        return -1;
+    anetCloexec(fds[0]);
+    anetCloexec(fds[1]);
```

Comment:
These are added syscalls for non-linux if O_CLOEXEC is not set.

To avoid doing extra syscalls, do this only if the flag is set in read_flags and write_flags.

Reply:
Done.